singer/src/main/java/com/pinterest/singer/reader/mapped/MappedThriftLogFileReader.java

```diff
@@ -0,0 +1,195 @@
+/**
+ * Copyright 2019 Pinterest, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.pinterest.singer.reader.mapped;
+
+import java.io.IOException;
+import java.nio.file.FileSystems;
+
+import org.apache.thrift.TException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import com.pinterest.singer.metrics.OpenTsdbMetricConverter;
+import com.pinterest.singer.reader.LogFileReader;
+import com.pinterest.singer.reader.LogFileReaderException;
+import com.pinterest.singer.thrift.LogFile;
+import com.pinterest.singer.thrift.LogMessage;
+import com.pinterest.singer.thrift.LogMessageAndPosition;
+import com.pinterest.singer.thrift.LogPosition;
+import com.pinterest.singer.utils.SingerUtils;
+import com.twitter.ostrich.stats.Stats;
+
+/**
+ * Reader that reads from thrift LogFile.
+ * 
+ * USE THIS READER FOR BATCH LOADING DATA ONLY, NOT FOR REAL-TIME STREAMING.
+ * <p>
+ * This class is NOT thread-safe.
+ */
+public class MappedThriftLogFileReader implements LogFileReader {
+
+  private static final Logger LOG = LoggerFactory.getLogger(MappedThriftLogFileReader.class);
+
+  // Factory that create LogMessage thrift objects.
+  private static final class LogMessageFactory
+      implements MappedThriftReader.TBaseFactory<com.pinterest.singer.thrift.LogMessage> {
```

Comment:
Does `LogMessage` need to be fully qualified here and below, since it's imported above?

Reply:
Fixed